### PR TITLE
RTEMIS-66: Added the campaign mini node.

### PR DESCRIPTION
**Pull request by kunal-sin**
Ticket # : https://innoraft.atlassian.net/browse/RTEMIS-66

---
### What the PR does
`(Explain fixes, changes and impacts)`
- Added the campaign mini node.
- Configured campaign permission for state admin.

### CODE REVIEW

File: modules/rte_mis_state/config/install/eck.eck_type.mini_node.campaign.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+langcode: en
+status: true
+dependencies:
+  config:
+    - eck.eck_entity_type.mini_node
+name: Campaign
+type: campaign
+description: ''
```
Comment (jaswinsingh) [inline modules/rte_mis_state/config/install/eck.eck_type.mini_node.campaign.yml:8]: Can we add description here?

File: modules/rte_mis_state/config/install/field.field.mini_node.campaign.field_academic_year.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    - eck.eck_type.mini_node.campaign
+    - field.storage.mini_node.field_academic_year
+  module:
+    - options
+id: mini_node.campaign.field_academic_year
+field_name: field_academic_year
+entity_type: mini_node
+bundle: campaign
+label: 'Academic Year'
+description: ''
```
Comment (jaswinsingh) [inline modules/rte_mis_state/config/install/field.field.mini_node.campaign.field_academic_year.yml:14]: Can we add description here.

File: modules/rte_mis_state/config/install/field.field.mini_node.campaign.field_campaign_items.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+    - field.storage.mini_node.field_campaign_items
+    - paragraphs.paragraphs_type.date_range
+  module:
+    - entity_reference_revisions
+id: mini_node.campaign.field_campaign_items
+field_name: field_campaign_items
+entity_type: mini_node
+bundle: campaign
+label: 'Campaign Items'
+description: ''
```
Comment (jaswinsingh) [inline modules/rte_mis_state/config/install/field.field.mini_node.campaign.field_campaign_items.yml:15]: Can we add description here?

File: modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_date.yml [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    - field.storage.paragraph.field_date
+    - paragraphs.paragraphs_type.date_range
+  module:
+    - datetime_range
+id: paragraph.date_range.field_date
+field_name: field_date
+entity_type: paragraph
+bundle: date_range
+label: Date
+description: ''
```
Comment (jaswinsingh) [inline modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_date.yml:14]: Can we add description here?

File: modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_event_type.yml [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    - field.storage.paragraph.field_event_type
+    - paragraphs.paragraphs_type.date_range
+  module:
+    - options
+id: paragraph.date_range.field_event_type
+field_name: field_event_type
+entity_type: paragraph
+bundle: date_range
+label: 'Event Type'
+description: ''
```
Comment (jaswinsingh) [inline modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_event_type.yml:14]: Can we add description here?

File: modules/rte_mis_state/config/install/eck.eck_type.mini_node.campaign.yml (hunk 1 above)
Comment (kunal-sin) [reply modules/rte_mis_state/config/install/eck.eck_type.mini_node.campaign.yml]: Added!

File: modules/rte_mis_state/config/install/field.field.mini_node.campaign.field_academic_year.yml (hunk 2 above)
Comment (kunal-sin) [reply modules/rte_mis_state/config/install/field.field.mini_node.campaign.field_academic_year.yml]: Added!

File: modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_date.yml (hunk 4 above)
Comment (kunal-sin) [reply modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_date.yml]: Added!

File: modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_event_type.yml (hunk 5 above)
Comment (kunal-sin) [reply modules/rte_mis_state/config/install/field.field.paragraph.date_range.field_event_type.yml]: Added!

**Review by jaswinsingh — APPROVED** [review]

**Pull request merged**